crates/primitives/receipt/src/from_blockifier.rs

```diff
@@ -82,8 +79,13 @@ pub fn from_blockifier_execution_info(res: &TransactionExecutionInfo, tx: &Trans
         _ => None,
     };
 
-    let messages_sent = res
-        .non_optional_call_infos()
+    fn recursive_call_info_iter(res: &TransactionExecutionInfo) -> impl Iterator<Item = &CallInfo> {
+        res
+            .non_optional_call_infos() // all root callinfos
+            .flat_map(|call_info| call_info.iter()) // flatmap over the roots' recursive inner call infos
```

Comment:
Are you sure this is recursive? Looking at the source code in Blockifier, it seems that `CallInfo::iter` indeed yields an iterator over its `inner_calls: Vec<CallInfo>`, but this does not recurse to return the `inner_calls` of _these_ `CallInfo`s.

This means the iterator will work but we will be missing inner call infos which are more than 1 level of recursion deep. This might be why this pr does not pass the kkrt tests.

Reply:
See https://docs.rs/blockifier/latest/src/blockifier/execution/call_info.rs.html#178-181
The `iter` in CallInfo returns a CallInfoIter, whose inner vec queue starts with `self` in it.
The CallInfoIter https://docs.rs/blockifier/latest/src/blockifier/execution/call_info.rs.html#229 here implements Iterator by popping the last call info from its queue, adding the child call infos to the queue, and returning the popped call info. This effectively implements a Breadth first search algorithm into the call info tree, as an Iterator returning &CallInfo. This goes into all of the children inner calls.

Reply:
Okay yep I got it, thanks for the explenation :+1: